elisp/guix-deriviation.el

```diff
@@ -0,0 +1,89 @@
+;;; guix-deriviation.el --- Guix deriviation mode
+
+;; Copyright © 2017 Oleg Pykhalov <go.wigust@gmail.com>
+;; Copyright © 2017 Alex Kost <alezost@gmail.com>
+
+;; This file is part of Emacs-Guix.
+
+;; Emacs-Guix is free software; you can redistribute it and/or modify
+;; it under the terms of the GNU General Public License as published by
+;; the Free Software Foundation, either version 3 of the License, or
+;; (at your option) any later version.
+;;
+;; Emacs-Guix is distributed in the hope that it will be useful,
+;; but WITHOUT ANY WARRANTY; without even the implied warranty of
+;; MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+;; GNU General Public License for more details.
+;;
+;; You should have received a copy of the GNU General Public License
+;; along with Emacs-Guix.  If not, see <http://www.gnu.org/licenses/>.
+
+;;; Commentary:
+
+;; This package provides minor-mode for prettifying Guix deriviation files.
+
+;;; Code:
+
+(require 'guix-utils)
+
+(defgroup guix-deriviation nil
+  "Prettify Guix deriviation files."
+  :prefix "guix-deriviation-file-"
+  :group 'guix
+  :group 'convenience)
+
+(defcustom guix-deriviation-file-regexp-group 1
+  "Regexp group in `guix-prettify-regexp' for prettifying."
+  :type 'integer
+  :group 'guix-deriviation)
+
+(defcustom guix-deriviation-file-regexp
+  (rx (group (and "/gnu" "/store")
+             ;; Hash-parts do not include "e", "o", "u" and "t".  See base32Chars
+             ;; at <https://github.com/NixOS/nix/blob/master/src/libutil/hash.cc>
+             "/" (= 32 (any "0-9" "a-d" "f-n" "p-s" "v-z"))
+             (* (any "-" "." "0-9" "a-z")) ".drv"))
```

Comment:
You missed "+" and "_" characters, so file names like these:

    /gnu/store/v1lil81w3a1hp5j4hrjhmz8am7c9g7s8-ld-wrapper-x86_64-guix-linux-gnu-0.drv
    /gnu/store/kvhvj09xa3rx7d15npgndcf4lg2prqpc-libstdc++-4.9.4.drv

would not be "buttonized".  Overall, I would write it like this:

```elisp
  (rx "/gnu/store/"
      (= 32 (any "0-9" "a-d" "f-n" "p-s" "v-z"))
      (+ (any "-_+." alnum)) ".drv")
```

Actually, the hash part `(= 32 ...)` should be generalized since
it is used by `guix-prettify-regexp` but I will do it later.